addons/mail/static/src/discuss/core/common/thread_service_patch.js

```diff
@@ -47,4 +47,25 @@ patch(ThreadService.prototype, {
             thread.isLoadingAttachments = false;
         }
     },
+
+    async muteThread(thread, { minutes = false } = {}) {
+        const muteUntil = await this.rpc("/discuss/channel/mute", {
+            channel_id: thread.id,
+            minutes,
+        });
+        thread.update({
+            notification_mute_until_dt: muteUntil,
+        });
+    },
+
+    async updateNotificationSetting(thread, notificationType) {
+        // Update the UI instantly to provide a better UX (no need to wait for the RPC to finish).
+        thread.update({
+            notificationType,
+        });
```

Comment:
Again avoid `update()` for the time being.
Probably later it'll be ok to use it, but currently there's too much custom code in there, and there's ongoing work to remove the complexity. By using it now, it makes this ongoing work harder to remove, so better not add more uses at the moment.

Instead, simply assign the value rather than `.update()`. Probably `update()` will become like a plain assign, so it's gonna be a syntatic sugar for dryer code. But that's not yet the case!